net/net_linux.go

```diff
@@ -401,40 +401,54 @@ func ConnectionsMaxWithContext(ctx context.Context, kind string, max int) ([]Con
 	return ConnectionsPidMax(kind, 0, max)
 }
 
+// Return a list of network connections opened, omitting `Uids`.
+func ConnectionsWithoutUids(kind string) ([]ConnectionStat, error) {
+	return ConnectionsWithoutUidsWithContext(context.Background(), kind)
+}
+
+func ConnectionsWithoutUidsWithContext(ctx context.Context, kind string) ([]ConnectionStat, error) {
+	return ConnectionsMaxWithoutUidsWithContext(ctx, kind, 0)
+}
+
+func ConnectionsMaxWithoutUidsWithContext(ctx context.Context, kind string, max int) ([]ConnectionStat, error) {
+	return ConnectionsPidMaxWithoutUidsWithContext(ctx, kind, 0, max)
+}
+
 // Return a list of network connections opened by a process.
 func ConnectionsPid(kind string, pid int32) ([]ConnectionStat, error) {
 	return ConnectionsPidWithContext(context.Background(), kind, pid)
 }
 
+func ConnectionsPidWithoutUids(kind string, pid int32) ([]ConnectionStat, error) {
+	return ConnectionsPidWithoutUidsWithContext(context.Background(), kind, pid)
+}
+
 func ConnectionsPidWithContext(ctx context.Context, kind string, pid int32) ([]ConnectionStat, error) {
-	tmap, ok := netConnectionKindMap[kind]
-	if !ok {
-		return nil, fmt.Errorf("invalid kind, %s", kind)
-	}
-	root := common.HostProc()
-	var err error
-	var inodes map[string][]inodeMap
-	if pid == 0 {
-		inodes, err = getProcInodesAll(root, 0)
-	} else {
-		inodes, err = getProcInodes(root, pid, 0)
-		if len(inodes) == 0 {
-			// no connection for the pid
-			return []ConnectionStat{}, nil
-		}
-	}
-	if err != nil {
-		return nil, fmt.Errorf("cound not get pid(s), %d: %s", pid, err)
-	}
-	return statsFromInodes(root, pid, tmap, inodes)
+	return ConnectionsPidMaxWithoutUidsWithContext(ctx, kind, pid, 0)
+}
+
+func ConnectionsPidWithoutUidsWithContext(ctx context.Context, kind string, pid int32) ([]ConnectionStat, error) {
+	return ConnectionsPidMaxWithoutUidsWithContext(ctx, kind, pid, 0)
 }
 
 // Return up to `max` network connections opened by a process.
 func ConnectionsPidMax(kind string, pid int32, max int) ([]ConnectionStat, error) {
 	return ConnectionsPidMaxWithContext(context.Background(), kind, pid, max)
 }
 
+func ConnectionsPidMaxWithoutUids(kind string, pid int32, max int) ([]ConnectionStat, error) {
+	return ConnectionsPidMaxWithoutUidsWithContext(context.Background(), kind, pid, max)
+}
+
 func ConnectionsPidMaxWithContext(ctx context.Context, kind string, pid int32, max int) ([]ConnectionStat, error) {
+	return connectionsPidMaxWithoutUidsWithContext(ctx, kind, pid, max, false)
+}
+
+func ConnectionsPidMaxWithoutUidsWithContext(ctx context.Context, kind string, pid int32, max int) ([]ConnectionStat, error) {
+	return connectionsPidMaxWithoutUidsWithContext(ctx, kind, pid, max, true)
```

Comment:
This invoked *without* Uids. So this becomes incompatible from current. Current returns with uids.
```suggestion
	return connectionsPidMaxWithoutUidsWithContext(ctx, kind, pid, max, false)
```

Reply:
I chose a negative `skipUids` for the 5th parameter of `connectionsPidMaxWithoutUidsWithContext`, so that `false` would be the original implementation of not skipping Uids. `true` causes the `Uids` fields not to be populated. When `false` the `Uids` field will be populated or an empty slice. When `true` the field will always be a nil slice.

I had a mistake where `ConnectionsPidWithContext` was calling a `ConnectionsPidMaxWithoutUidsWithContext`. I have corrected that to call `ConnectionsPidMaxWithContext`.

Here is a small proof of work:
```shell
# cat test/main.go 
package main

import (
	"github.com/shirou/gopsutil/net"
	"fmt"
)

func main() {
	v, _ := net.Connections("inet")
	fmt.Printf("Connections: %s\n", v)
	v2, _ := net.ConnectionsWithoutUids("inet")
	fmt.Printf("ConnectionsWithoutUids: %s\n", v2)
}
# go run test/main.go 
Connections: [{"fd":0,"family":2,"type":1,"localaddr":{"ip":"0.0.0.0","port":58841},"remoteaddr":{"ip":"0.0.0.0","port":0},"status":"LISTEN","uids":[],"pid":0} {"fd":0,"family":2,"type":1,"localaddr":{"ip":"0.0.0.0","port":111},"remoteaddr":{"ip":"0.0.0.0","port":0},"status":"LISTEN","uids":[],"pid":0} {"fd":0,"family":10,"type":1,"localaddr":{"ip":"::","port":42509},"remoteaddr":{"ip":"::","port":0},"status":"LISTEN","uids":[],"pid":0} {"fd":0,"family":10,"type":1,"localaddr":{"ip":"::","port":111},"remoteaddr":{"ip":"::","port":0},"status":"LISTEN","uids":[],"pid":0} {"fd":0,"family":10,"type":1,"localaddr":{"ip":"::","port":8080},"remoteaddr":{"ip":"::","port":0},"status":"LISTEN","uids":[],"pid":0} {"fd":0,"family":2,"type":2,"localaddr":{"ip":"127.0.0.1","port":996},"remoteaddr":{"ip":"0.0.0.0","port":0},"status":"NONE","uids":[],"pid":0} {"fd":0,"family":2,"type":2,"localaddr":{"ip":"0.0.0.0","port":111},"remoteaddr":{"ip":"0.0.0.0","port":0},"status":"NONE","uids":[],"pid":0} {"fd":0,"family":2,"type":2,"localaddr":{"ip":"0.0.0.0","port":40170},"remoteaddr":{"ip":"0.0.0.0","port":0},"status":"NONE","uids":[],"pid":0} {"fd":0,"family":10,"type":2,"localaddr":{"ip":"::","port":40020},"remoteaddr":{"ip":"::","port":0},"status":"NONE","uids":[],"pid":0} {"fd":0,"family":10,"type":2,"localaddr":{"ip":"::","port":111},"remoteaddr":{"ip":"::","port":0},"status":"NONE","uids":[],"pid":0}]
ConnectionsWithoutUids: [{"fd":0,"family":2,"type":1,"localaddr":{"ip":"0.0.0.0","port":58841},"remoteaddr":{"ip":"0.0.0.0","port":0},"status":"LISTEN","uids":null,"pid":0} {"fd":0,"family":2,"type":1,"localaddr":{"ip":"0.0.0.0","port":111},"remoteaddr":{"ip":"0.0.0.0","port":0},"status":"LISTEN","uids":null,"pid":0} {"fd":0,"family":10,"type":1,"localaddr":{"ip":"::","port":42509},"remoteaddr":{"ip":"::","port":0},"status":"LISTEN","uids":null,"pid":0} {"fd":0,"family":10,"type":1,"localaddr":{"ip":"::","port":111},"remoteaddr":{"ip":"::","port":0},"status":"LISTEN","uids":null,"pid":0} {"fd":0,"family":10,"type":1,"localaddr":{"ip":"::","port":8080},"remoteaddr":{"ip":"::","port":0},"status":"LISTEN","uids":null,"pid":0} {"fd":0,"family":2,"type":2,"localaddr":{"ip":"127.0.0.1","port":996},"remoteaddr":{"ip":"0.0.0.0","port":0},"status":"NONE","uids":null,"pid":0} {"fd":0,"family":2,"type":2,"localaddr":{"ip":"0.0.0.0","port":111},"remoteaddr":{"ip":"0.0.0.0","port":0},"status":"NONE","uids":null,"pid":0} {"fd":0,"family":2,"type":2,"localaddr":{"ip":"0.0.0.0","port":40170},"remoteaddr":{"ip":"0.0.0.0","port":0},"status":"NONE","uids":null,"pid":0} {"fd":0,"family":10,"type":2,"localaddr":{"ip":"::","port":40020},"remoteaddr":{"ip":"::","port":0},"status":"NONE","uids":null,"pid":0} {"fd":0,"family":10,"type":2,"localaddr":{"ip":"::","port":111},"remoteaddr":{"ip":"::","port":0},"status":"NONE","uids":null,"pid":0}]
```

Reply:
For context the original name for `connectionsPidMaxWithoutUidsWithContext` was `connectionsPidMaxWithContextUidsOpt`. I can change it back if that is a better name.